src/databricks_ai_bridge/model_serving_obo_credential_strategy.py

```diff
@@ -0,0 +1,111 @@
+import os
+import threading
+from typing import Dict, Optional, Tuple
+
+from databricks.sdk.core import Config
+from databricks.sdk.credentials_provider import (
+    CredentialsProvider,
+    CredentialsStrategy,
+    DefaultCredentials,
+)
+import logging
+
+logger = logging.getLogger(__name__)
+
+_MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH = "/var/credentials-secret/model-dependencies-oauth-token"
+
+def should_fetch_model_serving_environment_oauth() -> bool:
+        """
+        Check whether this is the model serving environment
+        Additionally check if the oauth token file path exists
+        """
+
+        is_in_model_serving_env = (
+            os.environ.get("IS_IN_DB_MODEL_SERVING_ENV")
+            or os.environ.get("IS_IN_DATABRICKS_MODEL_SERVING_ENV")
+            or "false"
+        )
+        return is_in_model_serving_env == "true" and os.path.isfile(
+            _MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH
+        )
+
+def _get_invokers_token():
+    main_thread = threading.main_thread()
+    thread_data = main_thread.__dict__
+    invokers_token = None
+    if "invokers_token" in thread_data:
+        invokers_token = thread_data["invokers_token"]
+
+    if invokers_token is None:
+        raise RuntimeError("Unable to read Invokers Token in Databricks Model Serving")
```

Comment:
```suggestion
        raise RuntimeError("Unable to read end user token in Databricks Model Serving. Please ...")
```

And make the error more actionable (complete the sentence after "Please ...")?